### PR TITLE
Start to refactor the MsSql Outbox to be a little cleaner

### PR DESCRIPTION
**Pull request by preardon**
The MSSQL Outbox has had a little bit of spaghetti code and quite a lot of duplications, the Aim of this PR is to clean the code and ensure that we are not duplicating code and that the same queries and methods are being used for complimentary paths

### CODE REVIEW

**Comment by preardon** [comment]
Managed to reduce the lines of code but a little over 200, also we now have more structure and less duplicated code.

Just a thought, we could probably Abstract class a decent portion of this and get it to service at least MySQL and Progress as well

let me know what you think

File: src/Paramore.Brighter.Outbox.MsSql/MsSqlOutbox.cs [hunk 1]
```diff
@@ -555,177 +295,207 @@ public void MarkDispatched(Guid id, DateTime? dispatchedAt = null, Dictionary<st
...
-            var sql = string.Format(pagingSqlFormat, _configuration.OutBoxTableName);
 
-            command.CommandText = sql;
-            command.Parameters.AddRange(parameters);
-        }
+        #endregion
 
-        private void CreatePagedReadCommand(SqlCommand command, int pageSize, int pageNumber)
+        #region Things that Actually call SQL
+        private void RunNonQuery(IAmABoxTransactionConnectionProvider transactionConnectionProvider, Func<SqlConnection, SqlCommand> commandFunc, Action loggingAction)
```
Comment (iancooper) [inline src/Paramore.Brighter.Outbox.MsSql/MsSqlOutbox.cs:311]: Isn't NonQuery a Command. This seems like a difficult negative naming choice

**Review by iancooper — COMMENTED** [review]
Looks a LOT better. I am slightly weirded by us surfacing NonQuery, it feels like we need a better name. Write vs. Read if no other option.

File: src/Paramore.Brighter.Outbox.MsSql/MsSqlOutbox.cs (hunk 1 above)
Comment (preardon) [reply src/Paramore.Brighter.Outbox.MsSql/MsSqlOutbox.cs:311]: Good spot, I can get that renamed

Comment (preardon) [reply src/Paramore.Brighter.Outbox.MsSql/MsSqlOutbox.cs]: I've renamed them ReadFromStore and WriteToStore Thought @ICooper ?

**Review by iancooper — APPROVED** [review]

**Pull request merged**